src/components/_classes/componentModal/ComponentModal.js

```diff
@@ -12,11 +12,11 @@ export default class ComponentModal {
     });
   }
 
-  constructor(component, modal, isOpened) {
+  constructor(component, modal, isOpened, currentValue) {
     this.isOpened = isOpened;
     this.component = component;
     this.modal = modal;
-    this.currentValue = this.component.dataValue;
+    this.currentValue = _.cloneDeep(currentValue);
```

Comment:
Can we use ```util.fastCloneDeep``` here?  Lodash clone deep is good for complex objects (but less performant), whereas fastCloneDeep is good for just cloning JSON objects.